crates/typst/src/diag.rs

```diff
@@ -509,6 +509,8 @@ pub enum PackageError {
     NetworkFailed(Option<EcoString>),
     /// The package archive was malformed.
     MalformedArchive(Option<EcoString>),
+    /// Some error when dealing with file locking.
+    FileLocking(EcoString),
```

Comment:
Honestly I think this can just be `Other`. It is very specific to the CLI and the `PackageError` is supposed to be generic over all Worlds, just distinguishing the most important stuff.

Reply:
One problem is that now it has a general file-locking error. Removing it will at least require expanding/rewriting concrete file-locking messages. Oh, since I have an error mapping function, I can just put string from this file to there. Should be easy to solve.

Reply:
It then has to be like this:

```
failed to load package (failed to open lock file: No such file or directory (os error 2))
```

Which, I'm not sure if it's pretty. I would prefer to use something like context from anyhow, so that all errors are one per line.